packages/block-editor/src/components/block-navigation/use-block-navigation-drop-zone.js

```diff
@@ -275,45 +189,85 @@ function getBlockNavigationDropTarget( blocksData, position ) {
 /**
  * A react hook for implementing a drop zone in block navigation.
  *
- * @param {Object} ref A React ref of a containing element for block navigation.
- *
  * @return {WPBlockNavigationDropZoneTarget} The drop target.
  */
-export default function useBlockNavigationDropZone( ref ) {
-	const [ target = {}, setTarget ] = useState();
+export default function useBlockNavigationDropZone() {
 	const {
-		rootClientId: targetRootClientId,
-		blockIndex: targetBlockIndex,
-	} = target;
+		getBlockRootClientId,
+		getBlockIndex,
+		getBlockCount,
+		getDraggedBlockClientIds,
+		canInsertBlocks,
+	} = useSelect( ( select ) => {
+		const selectors = select( blockEditorStore );
+		return {
+			canInsertBlocks: selectors.canInsertBlocks,
+			getBlockRootClientId: selectors.getBlockRootClientId,
+			getBlockIndex: selectors.getBlockIndex,
+			getBlockCount: selectors.getBlockCount,
+			getDraggedBlockClientIds: selectors.getDraggedBlockClientIds,
+		};
+	}, [] );
+	const [ target, setTarget ] = useState();
+	const { rootClientId: targetRootClientId, blockIndex: targetBlockIndex } =
+		target || {};
 
 	const dropEventHandlers = useOnBlockDrop(
 		targetRootClientId,
 		targetBlockIndex
 	);
 
-	const { position, type: dragEventType } = useDropZone( {
+	const ref = useDropZone( {
 		element: ref,
```

Comment:
Something is weird here, we pass a ref variable and we return it? 

Also, it seems like ideally we'd just remove the element argument right?

Reply:
Ah, good catch. The `element` and `withPosition` arguments can be removed, I simply forgot it.

Reply:
Removed in cb93fe7.